x-pack/legacy/plugins/siem/public/components/search_bar/index.tsx

```diff
@@ -0,0 +1,326 @@
+            : fromStr,
+          timelineId,
+        });
+      }
+
+      setFilterQuery({
+        id,
+        ...savedQueryUpdated.attributes.query,
+      });
+      setSavedQuery({ id, savedQuery: savedQueryUpdated });
+    };
+
+    const onClearSavedQuery = () => {
+      if (savedQuery != null) {
+        setFilterQuery({
+          id,
+          query: '',
+          language: savedQuery.attributes.query.language,
+        });
+        setSavedQuery({ id, savedQuery: undefined });
+        siemFilterManager.setFilters([]);
+      }
+    };
+
+    useEffect(() => {
+      let isSubscribed = true;
+      const subscriptions = new Subscription();
+
+      subscriptions.add(
+        siemFilterManager.getUpdates$().subscribe({
+          next: () => {
+            if (isSubscribed) {
+              setSearchBarFilter({
+                id,
+                filters: siemFilterManager.getFilters(),
+              });
+            }
+          },
+        })
+      );
+
+      return () => {
+        isSubscribed = false;
+        subscriptions.unsubscribe();
+      };
+    }, []);
+
+    return (
+      <SearchBarContainer data-test-subj={`${id}DatePicker`}>
+        <SearchBar
+          appName="siem"
+          isLoading={isLoading}
+          indexPatterns={[indexPattern as IndexPattern]}
+          query={filterQuery}
+          onClearSavedQuery={onClearSavedQuery}
+          onQuerySubmit={onQuerySubmit}
+          onRefresh={onRefresh}
+          onSaved={onSaved}
+          onSavedQueryUpdated={onSavedQueryUpdated}
+          savedQuery={savedQuery}
+          showFilterBar={true}
+          showDatePicker={true}
+          showQueryBar={true}
+          showQueryInput={true}
+          showSaveQuery={true}
+        />
+      </SearchBarContainer>
+    );
+  }
+);
+
+const makeMapStateToProps = () => {
+  const getEndSelector = endSelector();
+  const getFromStrSelector = fromStrSelector();
+  const getIsLoadingSelector = isLoadingSelector();
+  const getKindSelector = kindSelector();
+  const getQueriesSelector = queriesSelector();
+  const getStartSelector = startSelector();
+  const getToStrSelector = toStrSelector();
+  const getFilterQuerySelector = filterQuerySelector();
+  const getSavedQuerySelector = savedQuerySelector();
+  return (state: State, { id }: SiemSearchBarProps) => {
+    const inputsRange: InputsRange = getOr({}, `inputs.${id}`, state);
+    return {
+      end: getEndSelector(inputsRange),
+      fromStr: getFromStrSelector(inputsRange),
+      filterQuery: getFilterQuerySelector(inputsRange),
+      isLoading: getIsLoadingSelector(inputsRange),
+      kind: getKindSelector(inputsRange),
+      queries: getQueriesSelector(inputsRange),
+      savedQuery: getSavedQuerySelector(inputsRange),
+      start: getStartSelector(inputsRange),
+      toStr: getToStrSelector(inputsRange),
+    };
+  };
+};
+
+SearchBarComponent.displayName = 'SiemSearchBarComponent';
```

Comment:
May want this to match the component name to make searching easier when going from component tree -> ide.
```suggestion
SearchBarComponent.displayName = 'SearchBarComponent';
```